facilitator/src/task/pubsub.rs

```diff
@@ -90,29 +91,24 @@ impl<T: Task> TaskQueue<T> for GcpPubSubTaskQueue<T> {
             self.pubsub_api_endpoint, self.gcp_project_id, self.subscription_id
         );
 
-        let http_response = send_json_request(JsonRequestParameters {
-            request: ureq::post(&url),
+        let request = HTTP.prepare_request(RequestParameters {
+            url: &Url::parse(&url)?,
+            method: "POST",
             token_provider: Some(&mut self.oauth_token_provider),
-            // Empirically, if there are no messages available in the
-            // subscription, the PubSub API will wait about 90 seconds to send
-            // an HTTP 200 with an empty JSON body. We set higher timeouts than
-            // usual to allow for this.
-            connect_timeout_millis: Some(180_000), // three minutes
-            read_timeout_millis: Some(180_000),    // three minutes
```

Comment:
I think you will need to reintroduce the timeout parameters so that this particular HTTP request can wait longer than 10 seconds. Since timeouts are set on the `ureq::Agent`, I think this also supports my arguments against the global variable `HTTP` in `http.rs`.